src/realm/util/file.cpp

```diff
@@ -154,12 +154,22 @@ void make_dir(const std::string& path)
 
 void remove_dir(const std::string& path)
 {
+    if (try_remove_dir(path))
+        return;
+    int err = ENOENT;
+    std::string msg = get_errno_msg("remove() failed: ", err);
+    throw File::NotFound(msg, path);
+}
```

Comment:
I'd appreciate a `// Throws` on the line that calls `try_remove_dir()`. Similarly for `make_dir()`.